### PR TITLE
Fix: extra blocks added to database in realtime

### PR DESCRIPTION
**Pull request by kyscott18**

### CODE REVIEW

File: packages/core/src/sync-realtime/index.ts [hunk 1]
```diff
@@ -395,9 +411,24 @@ export const createRealtimeSync = (
         }
 
         return isMatched;
-      }),
+      });
+
+      for (let i = 0; i < callTraceFilters.length; i++) {
+        if (isFilterMatched[i]!) matchedFilters.add(callTraceFilters[i]!);
```
Comment (0xOlias) [inline packages/core/src/sync-realtime/index.ts:417]: Could move this into the map above

File: packages/core/src/sync/index.ts [hunk 2]
```diff
@@ -430,7 +430,10 @@ export const createSync = async (args: CreateSyncParameters): Promise<Sync> => {
             const chainId = network.chainId;
 
             await Promise.all([
-              args.syncStore.insertBlocks({ blocks: [event.block], chainId }),
+              args.syncStore.insertBlocks({
```
Comment (0xOlias) [inline packages/core/src/sync/index.ts:433]: Should we also conditionally call these sync store methods based on `event.filters` ? Or are they each smart enough to no-op if there is no data to insert?

Comment (kyscott18) [reply packages/core/src/sync/index.ts:433]: They each no-op and skip metrics if the data is empty

File: packages/core/src/sync-realtime/index.ts (hunk 1 above)
Comment (kyscott18) [reply packages/core/src/sync-realtime/index.ts:417]: After a second glance, with the way there are two separate branches in the block above for filters that have a factory or not, I think it is more readable how it is right now

**Pull request merged**